### PR TITLE
align arrange with `Ecto.Query.order_by/3`

### PR DESCRIPTION
**Pull request by cigrainger**
This changes `DataFrame.arrange/2` to match the `Ecto.Query.order_by/3` API as suggested by @josevalim.

### CODE REVIEW

File: lib/explorer/data_frame.ex [hunk 1]
```diff
@@ -631,32 +605,33 @@ defmodule Explorer.DataFrame do
   @spec arrange(
           df :: DataFrame.t(),
           columns ::
-            [String.t() | {String.t(), :asc | :desc}]
-            | Keyword.t()
+            String.t() | [String.t() | {:asc | :desc, String.t()}] | Keyword.t()
```
Comment (polvalente) [inline lib/explorer/data_frame.ex:608]:
```suggestion
            String.t() | [String.t() | {:asc | :desc, String.t()}]
```

You don't need the keyword() spec due to  it being equivalent to `[{atom(), term()}]`. Also, only strings are valid as values in this context

Comment (polvalente) [reply lib/explorer/data_frame.ex:608]: Also, I think that Keyword.t() doesn't exist

**Review by polvalente — APPROVED** [review]
Just some minor comments. If you make the suggested changes I can approve it again :)

File: lib/explorer/data_frame.ex (hunk 1 above)
Comment (cigrainger) [reply lib/explorer/data_frame.ex:608]: Thanks! `Keyword.t()` does exist: https://hexdocs.pm/elixir/1.12/Keyword.html#t:t/0. But you're right it's not necessary anymore!

Comment (cigrainger) [reply lib/explorer/data_frame.ex:608]: Only string should be valid, so that's good.

Comment (polvalente) [reply lib/explorer/data_frame.ex]: Nice! I though only keyword() was valid :)

**Review by polvalente — APPROVED** [review]

**Pull request merged**